libgnucash/engine/gnc-commodity.cpp

```diff
@@ -188,54 +188,75 @@ static QuoteSourceList currency_quote_sources =
  */
 static QuoteSourceList single_quote_sources =
 {
-    { false, SOURCE_SINGLE, "Alphavantage, US", "alphavantage" },
-    { false, SOURCE_SINGLE, "Amsterdam Euronext eXchange, NL", "aex" },
-    { false, SOURCE_SINGLE, "Association of Mutual Funds in India", "amfiindia" },
-    { false, SOURCE_SINGLE, "Australian Stock Exchange, AU", "asx" },
-    { false, SOURCE_SINGLE, "Canada Mutual", "canadamutual" },
-    { false, SOURCE_SINGLE, "Deka Investments, DE", "deka" },
-    { false, SOURCE_SINGLE, "Dutch", "dutch" },
-    { false, SOURCE_SINGLE, "DWS, DE", "dwsfunds" },
-    { false, SOURCE_SINGLE, "Financial Times Funds service, GB", "ftfunds" },
-    { false, SOURCE_SINGLE, "Finanzpartner, DE", "finanzpartner" },
-    { false, SOURCE_SINGLE, "GoldMoney spot rates, JE", "goldmoney" },
-    { false, SOURCE_SINGLE, "Google Web, US Stocks", "googleweb" },
-    { false, SOURCE_SINGLE, "India Mutual", "indiamutual" },
-    { false, SOURCE_SINGLE, "Morningstar, GB", "morningstaruk" },
-    { false, SOURCE_SINGLE, "Morningstar, JP", "morningstarjp" },
-    { false, SOURCE_SINGLE, "New Zealand stock eXchange, NZ", "nzx" },
-    { false, SOURCE_SINGLE, "Paris Stock Exchange/Boursorama, FR", "bourso" },
-    { false, SOURCE_SINGLE, "Romania", "romania" },
-    { false, SOURCE_SINGLE, "SIX Swiss Exchange shares, CH", "six" },
-    { false, SOURCE_SINGLE, "Skandinaviska Enskilda Banken, SE", "seb_funds" },
-    { false, SOURCE_SINGLE, "Sharenet, ZA", "za" },
-    { false, SOURCE_SINGLE, "TIAA-CREF, US", "tiaacref" },
-    { false, SOURCE_SINGLE, "Toronto Stock eXchange, CA", "tsx" },
-    { false, SOURCE_SINGLE, "T. Rowe Price", "troweprice" },
-    { false, SOURCE_SINGLE, "T. Rowe Price, US", "troweprice_direct" },
-    { false, SOURCE_SINGLE, "Union Investment, DE", "unionfunds" },
-    { false, SOURCE_SINGLE, "US Govt. Thrift Savings Plan", "tsp" },
-    { false, SOURCE_SINGLE, "Yahoo as JSON", "yahoo_json" },
-    { false, SOURCE_SINGLE, "Yahoo Web", "yahooweb" },
-    { false, SOURCE_SINGLE, "YH Finance (FinanceAPI)", "financeapi" },
+    { false, SOURCE_SINGLE, N_("Alphavantage"), "alphavantage" },
```

Comment:
IMHO we should then add a msgcontext, e.g.:
    { false, SOURCE_SINGLE, C_("FQ source", "Alphavantage"), "alphavantage" },
…

Reply:
Oops, we need a delay. I fear we have no macro for that defined, @jralls ?

Reply:
I know that the following is probably mudding waters even more, instead of driving to a solution, but I think it's even more subtle.

When I originally added Borsa Italiana I went to their home pages and [it doesn't seem they are encouraging translation of their name](https://www.borsaitaliana.it/borsaitaliana/borsaitaliana/italiana.en.htm). So: is it a trademark that should be left as is?

Some languages that don't use latin script may want to translate names that could be recognised as trademarks. One example could be Commerzbank that is rendered in Cyrillic script as Коммерцбанк. I know French, despite using latin script, also like to stick to their own language. But not sure how prolific it is in accounting applications translations.

There's even more. @bpschuck [mentioned](https://github.com/finance-quote/finance-quote/discussions/423#discussioncomment-10999079), that future versions of GnuCash would use dynamically just out of curiosity, how such strings that come from another application, could be translated?

Reply:
> When I originally added Borsa Italiana I went to their home pages and [it doesn't seem they are encouraging translation of their name](https://www.borsaitaliana.it/borsaitaliana/borsaitaliana/italiana.en.htm). So: is it a trademark that should be left as is?

The intro on that webpage says "Borsa Italiana has been founded after the privatisation of the exchange" and "from 29th April 2021, Borsa Italiana is part of Euronext Group." So I guess that it is indeed a company name. 

You have a good point about translating (or transliterating, but for our purposes there's no difference) for non-Latin scripts. That argues for marking all of them and providing a translator comment on the ones that are company names. I guess that might turn out to be all of them.

You're right that translating strings from another project requires that project to have a message catalog to retrieve the translations from and Finance::Quote at present doesn't. Using `methodinfo` in GnuCash will require a pretty big design change and (as @bpschuck pointed out in his reply) another piece of that redesign needs to be a way to handle obsolete sources. Let's keep this PR focused on the current design.